merqube_client_lib/session.py

```diff
@@ -167,23 +167,15 @@ def request(
         if url.startswith("//"):
             logger.warning(f"URL {url} starts with //, this is probably a mistake")
 
-        headers = dict(headers) if headers is not None else {}
-
-        # Generate a new requestid if this call isnt being made in a chain that already has it.
-        # (eg client calls dataapi, dataapi calls secapi - we dont want the second call to overwrite the original)
-        # if this isnt set by a client making a call to one of our APIs, the API itself will generate one (eg customer call)
-        # order is: 1) explicitly specified 2) set via chain, 3) generate new
-        headers[REQUEST_ID_HEADER] = headers.get(
-            REQUEST_ID_HEADER, os.getenv(MERQ_REQUEST_ID_ENV_VAR, str(uuid.uuid4()))
-        )
-
+        headers = headers or {}
```

Comment:
I think this was intentionally making a copy because `headers` is passed in by the client and modified below.

Reply:
ah good catch; ill add that back.